WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -1,2 +1,25 @@
 module CompetitionsHelper
+  def competition_message_for_user(competition, user)
+    message = ""
+    if registration = competition.registrations.find_by_user_id(user.id)
+      if registration.accepted?
+        message = "You are registered. "
+      else
+        message = "You are currently on the waiting list. "
+      end
+    end
+    if competition.isConfirmed
+      if competition.showAtAll
+        message += "This competition is confirmed and visible."
+      else
+        message += "This competition is confirmed but not visible."
+      end
```

Comment:
I like this pattern for these things.

    how = competition.showAtAll ? 'and' : 'but not'
    message += "This competition is confirmed #{how} visible."